File: app/src/main/java/com/example/wpossbank/UserProfileActivity.java
package com.example.wpossbank;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.database.Cursor;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.wpossbank.adaptadores.TransactionLogAdapter;
import com.example.wpossbank.database.Database;
import com.example.wpossbank.modelos.MakeMessages;
import com.example.wpossbank.modelos.SharedPreference;
import com.example.wpossbank.modelos.User;

import java.util.ArrayList;

import eightbitlab.com.blurview.BlurView;
import eightbitlab.com.blurview.RenderScriptBlur;

public class UserProfileActivity extends AppCompatActivity {
    Context context;
    SharedPreference sp;
    Database db;
    MakeMessages messages;

    User user;

    TextView balanceText;
    Button addMoneyButton;
    ImageView backArrow;

    BlurView blurView;

    TransactionLogAdapter adapter;
    RecyclerView recyclerView;

    ArrayList<String> entryDate, entryType, entrySource,entryValue;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_profile);

        context = this;
        sp = new SharedPreference(context);
        db = new Database(context);
        messages = new MakeMessages();

        user = new User(context);

        user.loadData(user);
        balanceText = findViewById(R.id.textView2);
        addMoneyButton = findViewById(R.id.addMoneyButton);
        backArrow = findViewById(R.id.backArrow);

        recyclerView = findViewById((R.id.recyclerView));

        blurView = findViewById(R.id.blurView);
        blurBackground();

        balanceText.setText(String.format("$%s", messages.separateNumberRight(
                Integer.toString(user.getBalance()), ".", 3)));

        entryDate = new ArrayList<>();
        entryType = new ArrayList<>();
        entryValue = new ArrayList<>();
        entrySource = new ArrayList<>();
        StoreDataInArrays();

        adapter = new TransactionLogAdapter(context, entryDate, entryType, entryValue, entrySource);

        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(context));

        addMoneyButton.setOnClickListener( showAddMoneyActivity ->
                startActivity(new Intent(context, DepositsActivity.class)));

        backArrow.setOnClickListener( goBack -> finish());
    }

    private void StoreDataInArrays(){
        Cursor fetch = db.fetchData(sp.getActiveUser(),db.getTable("log"),db.getColumn("active user"));

        if (fetch.getCount() > 0){
            while (fetch.moveToNext()){
                entryDate.add(fetch.getString(2));
                Log.d("TAG",fetch.getString(2));
                entryType.add(fetch.getString(3));
                entryValue.add(fetch.getString(4));
                entrySource.add(fetch.getString(5));
            }
        }

        fetch.close();
    }

    private void blurBackground(){
        float radius = 20f;

        View decorView = getWindow().getDecorView();
        ViewGroup rootView = decorView.findViewById(android.R.id.content);
        Drawable windowBackground = decorView.getBackground();

        blurView.setupWith(rootView)
                .setFrameClearDrawable(windowBackground)
                .setBlurAlgorithm(new RenderScriptBlur(this))
                .setBlurRadius(radius)
                .setBlurAutoUpdate(true)
                .setHasFixedTransformationMatrix(true);
    }
}

File: app/src/main/java/com/example/wpossbank/activities/UserProfileActivity.kt
package com.example.wpossbank.activities

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.wpossbank.R
import com.example.wpossbank.adapters.TransactionLogAdapter
import com.example.wpossbank.database.Database
import com.example.wpossbank.models.MakeMessages
import com.example.wpossbank.models.SharedPreference
import com.example.wpossbank.models.User
import eightbitlab.com.blurview.BlurView
import eightbitlab.com.blurview.RenderScriptBlur
import java.util.*

class UserProfileActivity : AppCompatActivity() {
    lateinit var context: Context
    lateinit var sp: SharedPreference
    lateinit var db: Database
    lateinit var messages: MakeMessages
    lateinit var user: User
    private lateinit var balanceText: TextView
    private lateinit var addMoneyButton: Button
    lateinit var backArrow: ImageView
    lateinit var blurView: BlurView
    private lateinit var adapter: TransactionLogAdapter
    private lateinit var recyclerView: RecyclerView
    private lateinit var entryDate: MutableList<String>
    private lateinit var entryType: MutableList<String>
    private lateinit var entrySource: MutableList<String>
    private lateinit var entryValue: MutableList<String>
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_user_profile)
        context = this
        sp = SharedPreference(context)
        db = Database(context)
        messages = MakeMessages()
        user = User()
        user.loadData(context)
        balanceText = findViewById(R.id.textView2)
        addMoneyButton = findViewById(R.id.addMoneyButton)
        backArrow = findViewById(R.id.backArrow)
        recyclerView = findViewById(R.id.recyclerView)
        blurView = findViewById(R.id.blurView)
        blurBackground()
        balanceText.text = String.format("$%s", messages.separateNumberRight(
                    user.balance.toString(), ".", 3) )
        entryDate = ArrayList()
        entryType = ArrayList()
        entryValue = ArrayList()
        entrySource = ArrayList()
        storeDataInArrays()
        adapter = TransactionLogAdapter(context, entryDate, entryType, entryValue, entrySource)
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(context)
        addMoneyButton.setOnClickListener{
            startActivity(
                Intent(context, DepositsActivity::class.java)
            )
        }
        backArrow.setOnClickListener { finish() }
    }

    private fun storeDataInArrays() {
        val fetch =
            db.fetchData(sp.activeUser, db.getTable("log"), db.getColumn("active user"))
        if (fetch.count > 0) {
            while (fetch.moveToNext()) {
                entryDate.add(fetch.getString(2))
                Log.d("TAG", fetch.getString(2))
                entryType.add(fetch.getString(3))
                entryValue.add(fetch.getString(4))
                entrySource.add(fetch.getString(5))
            }
        }
        fetch.close()
    }

    private fun blurBackground() {
        val radius = 20f
        val decorView = window.decorView
        val rootView = decorView.findViewById<ViewGroup>(android.R.id.content)
        val windowBackground = decorView.background
        blurView.setupWith(rootView)
            .setFrameClearDrawable(windowBackground)
            .setBlurAlgorithm(RenderScriptBlur(this))
            .setBlurRadius(radius)
            .setBlurAutoUpdate(true)
            .setHasFixedTransformationMatrix(true)
    }
}